app/lib/kafka.server.ts

```diff
@@ -0,0 +1,39 @@
+/*!
+ * Copyright © 2023 United States Government as represented by the
+ * Administrator of the National Aeronautics and Space Administration.
+ * All Rights Reserved.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+import { Kafka } from 'gcn-kafka'
+
+import { domain, getEnvOrDie, hostname } from './env.server'
+
+const client_id = getEnvOrDie('KAFKA_CLIENT_ID')
+const client_secret = getEnvOrDie('KAFKA_CLIENT_SECRET')
+const producer = new Kafka({
+  client_id,
+  client_secret,
+  domain,
+}).producer()
+
+process.on('SIGTERM', async () => {
+  await producer.disconnect()
+  process.exit(0)
+})
+
+export async function sendKafka(item: any, topic: string) {
+  await producer.connect()
```

Comment:
Please lift this out to module scope so that it is amortized across Lambda warm starts.

Reply:
Moving this up results in this error: 
![image](https://github.com/nasa-gcn/gcn.nasa.gov/assets/11023698/b6865b63-ea28-4684-8d42-52a992f36d63)

Reply:
One workaround is to use `memoizee(..., { promise: true })`. We have several examples of that in our code.

Or we could have esbuild compile to ESM for us so that top-level awaits are legal.

Reply:
Here is the change to switch to an ESM build:

```patch
diff --git a/esbuild.js b/esbuild.js
index 21a02fc0..c1b4a43d 100644
--- a/esbuild.js
+++ b/esbuild.js
@@ -12,11 +12,12 @@ const entryPoints = await glob(
  */
 const options = {
   bundle: true,
+  format: 'esm',
   entryPoints,
   logLevel: 'info',
   outdir: 'build',
   outbase: 'app',
-  outExtension: { '.js': '.cjs' },
+  outExtension: { '.js': '.mjs' },
   external: ['@aws-sdk/*', 'aws-sdk'],
   platform: 'node',
   target: ['node18'],
```

Reply:
I'll opt for the memoizee, I tried the ESM build and got errors about dynamic imports of libraries not being supported

Reply:
Yeah, CJS vs ESM is a mess.

Reply:
The `await producer.connect()` _must_ be at module scope. Please add that to the promise-returning function that is memoized.